helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -424,6 +446,20 @@ public void invoke(NotificationContext changeContext) throws Exception {
         List<CurrentState> currentStates = preFetch(_propertyKey);
         currentStateChangeListener.onStateChange(instanceName, currentStates, changeContext);
 
+      } else if (_changeType == CUSTOMIZED_STATE_ROOT) {
+        CustomizedStateRootChangeListener customizedStateRootChangeListener =
+            (CustomizedStateRootChangeListener) _listener;
+        String instanceName = PropertyPathConfig.getInstanceNameFromPath(_path);
+        customizedStateRootChangeListener.onCustomizedStateRootChange(instanceName,
```

Comment:
If prefetch, I guess we should get all the children for types and pass to the onCustomizedStateRootChange() method?